.github/workflows/hackage.yml

```diff
@@ -103,22 +104,15 @@ jobs:
           cd $(ls -d ./incoming/${{ matrix.package }}-*)
           echo "packages: . ../../* ../../plugins/*" > cabal.project
 
-      - name: "Add temporary needed allow-newer"
-        if: steps.get-hackage-version.outputs.exists != 'true'
-        run: |
-          # TODO: remove when not needed
-          cd $(ls -d ./incoming/${{ matrix.package }}-*)
-          echo "allow-newer: Chart-diagrams:diagrams-core, SVGFonts:diagrams-core," >> cabal.project
-
       - name: "Add temporary needed allow-newer for ghc-9.0"
-        if: steps.get-hackage-version.outputs.exists != 'true' && matrix.ghc == '9.0.1'
+        if: steps.get-hackage-version.outputs.exists != 'true' && (matrix.ghc == '9.0.1' || matrix.ghc == '9.0.2')
         run: |
           # TODO: remove when not needed
           cd $(ls -d ./incoming/${{ matrix.package }}-*)
           # For brittany
-          echo " brittany:base, brittany:ghc, brittany:ghc-boot-th, butcher:base, multistate:base, data-tree-print:base," >> cabal.project
-          # For floskell and stylish-haskell
-          echo " floskell:base, floskell:ghc-prim, stylish-haskell:Cabal,stylish-haskell:ghc-lib-parser," >> cabal.project
+          echo " butcher:base, multistate:base, data-tree-print:base," >> cabal.project
+          # For stylish-haskell
+          echo " stylish-haskell:Cabal,stylish-haskell:ghc-lib-parser,stylish-haskell:aeson" >> cabal.project
```

Comment:
Seems like the same reasoning from https://github.com/haskell/haskell-language-server/pull/2567/files#r779448472 to be documented

Seems like the sum of: https://github.com/haskell/haskell-language-server/pull/2567/files#r779448472 & https://github.com/haskell/haskell-language-server/pull/2567/files#r779120031 is enough to form a rough understanding.

But also `hackage` & `build` are a "deep cut" stuff for core maintainers, but even in that case doc may be beneficial to keep in the file.